keps/sig-node/4817-resource-claim-device-status/README.md

```diff
@@ -0,0 +1,614 @@
+}
+
+
+// AllocatedDeviceStatus contains the status of an allocated device, if the
+// driver chooses to report it. This may include driver-specific information.
+type AllocatedDeviceStatus struct {
+	// Request is the name of the request in the claim which caused this
+	// device to be allocated. Multiple devices may have been allocated
+	// per request.
+	//
+	// +required
+	Request string `json:"request" protobuf:"bytes,1,rep,name=request"`
+
+	// Driver specifies the name of the DRA driver whose kubelet
+	// plugin should be invoked to process the allocation once the claim is
+	// needed on a node.
+	//
+	// Must be a DNS subdomain and should end with a DNS domain owned by the
+	// vendor of the driver.
+	//
+	// +required
+	Driver string `json:"driver" protobuf:"bytes,2,rep,name=driver"`
+
+	// This name together with the driver name and the device name field
+	// identify which device was allocated (`<driver name>/<pool name>/<device name>`).
+	//
+	// Must not be longer than 253 characters and may contain one or more
+	// DNS sub-domains separated by slashes.
+	//
+	// +required
+	Pool string `json:"pool" protobuf:"bytes,3,rep,name=pool"`
+
+	// Device references one device instance via its name in the driver's
+	// resource pool. It must be a DNS label.
+	//
+	// +required
+	Device string `json:"device" protobuf:"bytes,4,rep,name=device"`
+
+	// Conditions contains the latest observation of the device's state.
+	// If the device has been configured according to the class and claim
+	// config references, the `Ready` condition should be True.
+	//
+	// +optional
+	// +listType=atomic
+	Conditions []metav1.Condition `json:"conditions" protobuf:"bytes,5,rep,name=conditions"`
+
+	// DeviceInfo contains Arbitrary driver-specific data.
+	//
+	// +optional
+	// +listType=atomic
+	DeviceInfo []runtime.RawExtension `json:"deviceInfo,omitempty" protobuf:"bytes,6,rep,name=deviceInfo"`
+
+    // NetworkDeviceInfo contains network-related information specific to the device.
+	//
+	// +optional
+	NetworkDeviceInfo NetworkDeviceInfo `json:"networkDeviceInfo,omitempty" protobuf:"bytes,7,rep,name=networkDeviceInfo"`
+}
+
+// NetworkDeviceInfo provides network-related details for the allocated device.
+// This information may be filled by drivers or other components to configure
+// or identify the device within a network context.
+type NetworkDeviceInfo struct {
+	// Interface specifies the name of the network interface associated with
+	// the allocated device. This might be the name of a physical or virtual
+	// network interface.
+	//
+	// +optional
+	Interface string `json:"interface,omitempty" protobuf:"bytes,1,rep,name=interface"`
+
+	// IPs lists the IP addresses assigned to the device's network interface.
+	// This can include both IPv4 and IPv6 addresses.
+	//
+	// +optional
+	IPs []string `json:"ips,omitempty" protobuf:"bytes,2,rep,name=ips"`
```

Comment:
Does it make sense to have a type here instead of string? Does mask, family add value? 

`type IP struct {
    address 
    mask
    family
}`

Reply:
I replaced this field with a `NetworkAddresses` one:
```golang
type NetworkDeviceInfo struct {
...
    // NetworkAddresses lists the network addresses assigned to the device's network interface.
    // This can include both IPv4 and IPv6 addresses.
    //
    // +optional
    NetworkAddresses []NetworkAddress `json:"networkAddresses,omitempty" protobuf:"bytes,2,rep,name=networkAddresses"`
...
}

// NetworkAddress provides a network address related details such as IP and Mask.
type NetworkAddress struct {
    // CIDR contains the network address in CIDR notation, which includes
    // both the address and the associated subnet mask.
    // e.g.: "192.0.2.0/24" for IPv4 and "2001:db8::/64" for IPv6.
    //
    // +required
    CIDR string `json:"cidr,omitempty" protobuf:"bytes,1,rep,name=cidr"`
}
```
For reference here is how the [NetworkAttachmentDefinition Status](https://github.com/k8snetworkplumbingwg/network-attachment-definition-client/blob/v1.7.3/pkg/apis/k8s.cni.cncf.io/v1/types.go#L103) looks like and here is for the [CNI Result](https://github.com/containernetworking/cni/blob/v1.2.3/pkg/types/100/types.go#L89).

Reply:
I switched it back to a single field
```golang
type NetworkDeviceData struct {
    ...
    // Addresses lists the network addresses assigned to the device's network interface.
    // This can include both IPv4 and IPv6 addresses.
    // The addresses are in the CIDR notation, which includes both the address and the
    // associated subnet mask.
    // e.g.: "192.0.2.0/24" for IPv4 and "2001:db8::/64" for IPv6.
    //
    // +optional
    // +listType=atomic
    Addresses []string `json:"addresses,omitempty" protobuf:"bytes,2,opt,name=addresses"`
    ...
}
```